Assets/Scripts/DialogueBubble.cs

```diff
@@ -0,0 +1,275 @@
+﻿using System.Collections;
+using System.Collections.Generic;
+using System.Text.RegularExpressions;
+using UnityEngine;
+using TMPro;
+
+public class DialogueBubble : MonoBehaviour
+{
+    enum TextEffect { None, Shaky, Wavy }
+    TextEffect activeEffect;
+
+    private TMP_Text Text;
+    private string CurrentText;
+
+    const string kAlphaCode = "<color=#00000000>";
+    const float kMaxTextTime = 0.1f;
+    public static float TextSpeed = 2;
+    private float originalTextSpeed;
+
+    CanvasGroup Group;
+
+    public float AngleMultiplier = 1.0f;
+    public float SpeedMultiplier = 1.0f;
+    public float CurveScale = 1.0f;
+    private bool hasTextChanged;
+    private bool isTextShaking = false;
+
+    // Contains animation data
+    private struct VertexAnim
+    {
+        public float angleRange;
+        public float angle;
+        public float speed;
+    }
+
+    void Awake()
+    {
+        Text = GetComponentInChildren<TMP_Text>();
+        activeEffect = TextEffect.None;
+        // Store original speed to restore after acceleration
+        originalTextSpeed = TextSpeed;
+    }
+
+    void Start()
+    {
+        Group = GetComponent<CanvasGroup>();
+        Group.alpha = 0;
+    }
+
+    void Update()
+    {
+        if (Input.GetKeyDown(KeyCode.J))
+        {
+            isTextShaking = !isTextShaking;
```

Comment:
I noticed that it only turns off the shaking after you initiate a new dialogue box, not during the current open dialogue box. Is this intentional?

Reply:
I think that's ok for now

Reply:
Yeah, right now I'm only showing the effect. In order to disable it, the coroutine has to be stopped but I didn't add that in since the shake effect will not be user activated but script activated

Reply:
Got it! Thanks for the clarification!